config/mariadb/config.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/crossplane-contrib/provider-jet-azure/config/common"
+)
+
+const (
+	mariadbServerPort = 3306
+)
+
+// Configure configures mariadb group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_mariadb_server", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.LateInitializer = config.LateInitializer{
+			IgnoredFields: []string{"ssl_enforcement", "storage_profile"},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforMariaDB", "servers", "name")
+		r.Sensitive.AdditionalConnectionDetailsFn = func(attr map[string]interface{}) (map[string][]byte, error) {
+			return map[string][]byte{
+				xpv1.ResourceCredentialsSecretUserKey:     []byte(fmt.Sprintf("%s@%s", attr["administrator_login"], attr["name"])),
+				xpv1.ResourceCredentialsSecretPasswordKey: []byte(attr["administrator_login_password"].(string)),
+				xpv1.ResourceCredentialsSecretEndpointKey: []byte(attr["fqdn"].(string)),
+				xpv1.ResourceCredentialsSecretPortKey:     []byte(strconv.Itoa(mariadbServerPort)),
+			}, nil
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_database", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/databases/database1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforMariaDB",
+			"servers", "server_name",
+			"databases", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_firewall_rule", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/firewallRules/rule1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforMariaDB",
+			"servers", "server_name",
+			"firewallRules", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_virtual_network_rule", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+			"subnet_id": config.Reference{
```

Comment:
nit: We should be good without an explicit referencer for `subnet_id` configured here, as `subnet_id` is a common referencer field declared [here](https://github.com/crossplane-contrib/provider-jet-azure/blob/45e9f598d288b6ebd5eaf1ee62830482eec70b71/config/common/common.go#L153). Could you please give it a try?

Reply:
Confirming that it properly picked up `subnet_id` without the explicit configuration

```
    Subnet Id:    /subscriptions/<redacted>/resourceGroups/mariadb-test-yury/providers/Microsoft.Network/virtualNetworks/example-vn-yury/subnets/example-subnet-yury
```